### ISSUE TITLE
Allow using environment variables

### ISSUE THREAD

**Issue by rastytheamateur** [pawamoy/django-appsettings#64]
Environment variables are more and more popular, for example in kubernetes. It would be nice to be able to use them in the django-app-settings as well.

### Implementation description
Every `Setting` with an attribute defining the environment name, lets call it `environ_name`, would try to inherit the setting value from `os.environ`. If not set, take the normal value from `DJANGO_SETTINGS_MODULE`. If even this is not set, take the default value from `default` attribute.

This environ variable value would be passed through all validators.

For transforming the value we would add a boolean `transform_environ` property, which indicates whether to pass the environ variable value to the `transform(v)` method or not.

### Usage
```
# os.environ["FOO"] = "bar_env"
# settings.FOO = "bar_django"

s = Setting(name="foo", environ_name="foo", default="bar_def")
s.value == "bar_env"
```

```
# del os.environ["FOO"]
# settings.FOO = "bar_django"

s = Setting(name="foo", environ_name="foo", default="bar_def")
s.value == "bar_django"
```

```
# del os.environ["FOO"]
# del settings.FOO

s = Setting(name="foo", environ_name="foo", default="bar_def")
s.value == "bar_def"
```

**Comment by ziima** [comment]
I wouldn't add a new attribute. Instead I'd check environment variable for all settings.

Is there a use case for `transform_environ`?

**Comment by rastytheamateur** [comment]
> I wouldn't add a new attribute. Instead I'd check environment variable for all settings.

So you would actually require the django name and environ name to be the same?

> Is there a use case for `transform_environ`?

You may not want to transform the environ value, but the django value yes, or the other way around. It is just a possibility to distinguish the behavior between environ values and django values.

So your option is to consider django value and environ value the same? The only change would be that `row_value` would be the one from environ (if set) or from django settings?

**Comment by ziima** [comment]
> So you would actually require the django name and environ name to be the same?

Yes. More over, I would avoid having the different.

> You may not want to transform the environ value, but the django value yes, or the other way around. 

I don't really see a use case, apart from type conversion, since all environment variables have string values.

> So your option is to consider django value and environ value the same? The only change would be that raw_value would be the one from environ (if set) or from django settings?

Basically yes.

**Comment by rastytheamateur** [comment]
Well, I agree with your design.

> ... apart from type conversion ...

How to manage it? Every subclass of `Setting` will have a method for converting environ values? (I guess more complex settings like `ListSetting` and similar will not have this environ option?)

**Comment by ziima** [comment]
I'd add `decode(value: Any) -> Any` method to a `Setting` object to convert value from environment to `raw_value`. Basic implementation could just return the value and where required an override would be implemented. `ListSetting` could use a `:` as a separator, similarily to `PYTHONPATH`.

**Comment by rastytheamateur** [comment]
All right, gonna do it in the evening or tomorrow.

**Comment by ziima** [comment]
Fixed in #65.
